simulator/tests/test_config_parser.py

```diff
@@ -1,69 +1,106 @@
+        is_valid, errors = validate_config(yaml.safe_load(file))
+        assert is_valid == False
+        assert errors == {
+            "index_clean_up_age_days": ["required field"],
+            "index_roll_over_size": ["required field"],
+        }
 
 
 def test_validate_config_invalid_data_type():
-    is_valid, errors = validate_config(
-        yaml.safe_load(open("./config_test/config_5_F.yaml", "r"))
-    )
-    assert is_valid == False
-    assert errors == {
-        "data_ingestion": [
-            {
-                "states": [
-                    {3: [{"ingestion_rate_gb_per_hr": ["must be of number type"]}]}
-                ]
-            }
-        ]
-    }
+    with open(os.path.join(test_path, "config_5_F.yaml"), "r") as file:
+        is_valid, errors = validate_config(yaml.safe_load(file))
+        assert is_valid == False
+        assert errors == {
+            "data_ingestion": [
+                {
+                    "states": [
+                        {3: [{"ingestion_rate_gb_per_hr": ["must be of number type"]}]}
+                    ]
+                }
+            ]
+        }
 
 
 def test_validate_config_missing_nested_key():
-    is_valid, errors = validate_config(
-        yaml.safe_load(open("./config_test/config_6_F.yaml", "r"))
+    with open(os.path.join(test_path, "config_6_F.yaml"), "r") as file:
+        is_valid, errors = validate_config(yaml.safe_load(file))
+        assert is_valid == False
+        assert errors == {"searches": [{0: [{"probability": ["required field"]}]}]}
+
+
+def test_parse_config():
+    fp = open(os.path.join(test_path, "config_1_P.yaml"), "r")
+    all_configs = yaml.safe_load(fp.read())
+
+    data_generation_interval_minutes = all_configs.pop(
+        const.DATA_GENERATION_INTERVAL_MINUTES
+    )
+    data_ingestion = all_configs.pop(const.DATA_INGESTION)
+    searches = all_configs.pop(const.SEARCHES)
+    stats = all_configs
+
+    expected_config = Config(
+        stats, data_ingestion, searches, data_generation_interval_minutes
+    )
+    config = parse_config(os.path.join(test_path, "config_1_P.yaml"))
+    assert (
+        expected_config.data_generation_interval_minutes
+        == config.data_generation_interval_minutes
     )
-    assert is_valid == False
-    assert errors == {"searches": [{0: [{"probability": ["required field"]}]}]}
+    assert expected_config.stats == config.stats
+    assert expected_config.data_ingestion == config.data_ingestion
+    assert expected_config.searches == config.searches
+
+
+def test_parse_config_error_reading_config():
+    with pytest.raises(ValidationError) as e:
+        parse_config(os.path.join(test_path, "config_5_F.yaml"))
+        assert "error reading config file - " == e
+
+
+def test_parse_config_validate_error():
+    with pytest.raises(ValidationError) as e:
+        parse_config(os.path.join(test_path, "config_4_F.yaml"))
+        assert "Error validating config file - " == e

```

Comment:
Please add a doc-strings and comments for each test case explaining the purpose of the test case and also explain steps if they are a bit complex